@here/harp-materials/lib/ShaderChunks/ExtrusionChunks.ts

```diff
@@ -4,15 +4,25 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import { ExtrusionFeatureDefs } from "../MapMeshMaterialsDefs";
+
+const MIN_BUILDING_HEIGHT_SQUARED =
+    ExtrusionFeatureDefs.MIN_BUILDING_HEIGHT * ExtrusionFeatureDefs.MIN_BUILDING_HEIGHT;
+
 export default {
     extrusion_pars_vertex: `
 // Extrusion axis (xyz: vector, w: factor).
 attribute vec4 extrusionAxis;
 uniform float extrusionRatio;
 varying vec4 vExtrusionAxis;
+varying float vExtrusionRatio;
+
 `,
     extrusion_vertex: `
-transformed = transformed + extrusionAxis.xyz * (extrusionRatio - 1.0);
+const float MIN_BUILDING_HEIGHT_SQUARED = ${MIN_BUILDING_HEIGHT_SQUARED};
+float extrusionAxisLenSquared = dot(extrusionAxis.xyz, extrusionAxis.xyz);
+vExtrusionRatio = mix(1.0, extrusionRatio, float(extrusionAxisLenSquared > MIN_BUILDING_HEIGHT_SQUARED));
```

Comment:
Using float cast of bool expression with interpolation does not seem to be common and effective way, I would rather consider using OGLS step() function instead. Something like:
`vExtrusionRatio = max(1 - step(extrusionAxisLenSquared, MIN_BUILDING_HEIGHT_SQUARED), extrusionRatio)`
should give you same results but in more gentle manner.

Reply:
Done